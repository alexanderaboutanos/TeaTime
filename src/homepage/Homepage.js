/** @format */

import React, { useContext } from "react";
import { Link } from "react-router-dom";
import "./Homepage.css";
import UserContext from "../auth/UserContext";
import Background from "../common/Background";

/** Homepage of site.
 *
 * Shows welcome message or login/register buttons.
 *
 * Routed at /
 *
 * Routes -> Homepage
 */

function Homepage() {
  const { currentUser } = useContext(UserContext);
  console.debug("Homepage", "currentUser=", currentUser);

  return (
    <div className="Homepage">
      <Background />
      <div className="container text-center" style={{ zIndex: 100 }}>
        <h1 className="mb-4 font-weight-bold">TeaTime</h1>
        <p className="lead">All the teas in one, convenient place.</p>
        {currentUser ? (
          <h3>Welcome back, {currentUser.first_name}!</h3>
        ) : (
          <p>
            <Link className="btn btn-primary font-weight-bold mr-3" to="/login">
              Log in
            </Link>
            <Link className="btn btn-primary font-weight-bold" to="/signup">
              Sign up
            </Link>
          </p>
        )}
      </div>
    </div>
  );
}

export default Homepage;
