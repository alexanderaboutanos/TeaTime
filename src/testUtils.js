/** @format */

import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser1",
  first_name: "Test",
  last_name: "User",
};

const UserProvider = ({ children, currentUser = demoUser }) => (
  <UserContext.Provider value={{ currentUser }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
