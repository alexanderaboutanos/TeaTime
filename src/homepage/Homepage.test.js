/** @format */

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Homepage from "./Homepage";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders the proper homepage when logged out", function () {
  const { getByText, queryByText } = render(
    <MemoryRouter>
      <UserProvider currentUser={null}>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );

  // renders title and subtitle
  const h1 = getByText("TeaTime");
  const subtitle = getByText("All the teas in one, convenient place.");
  expect(h1).toBeInTheDocument();
  expect(subtitle).toBeInTheDocument();

  // renders login and signup button
  const loginBtn = getByText("Log in");
  const signupBtn = getByText("Sign up");
  expect(loginBtn).toBeInTheDocument();
  expect(signupBtn).toBeInTheDocument();

  // should not render the welcome message
  const welcomeMsg = queryByText("Welcome back,", { exact: false });
  expect(welcomeMsg).not.toBeInTheDocument();
});

it("renders the proper homepage when logged in", function () {
  const { getByText, queryByText } = render(
    <MemoryRouter>
      <UserProvider>
        <Homepage />
      </UserProvider>
    </MemoryRouter>
  );

  // renders title and subtitle
  const h1 = getByText("TeaTime");
  const subtitle = getByText("All the teas in one, convenient place.");
  expect(h1).toBeInTheDocument();
  expect(subtitle).toBeInTheDocument();

  // does not render login and signup button
  const loginBtn = queryByText("Log in");
  const signupBtn = queryByText("Sign up");
  expect(loginBtn).not.toBeInTheDocument();
  expect(signupBtn).not.toBeInTheDocument();

  // should render the welcome message
  const welcomeMsg = queryByText("Welcome back,", { exact: false });
  expect(welcomeMsg).toBeInTheDocument();
});
