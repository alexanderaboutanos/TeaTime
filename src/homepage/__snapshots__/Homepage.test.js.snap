// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <img
      alt="teacup"
      src="https://images.unsplash.com/photo-1503933190776-da31c7700f80?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=3307&q=80"
      style="height: 100%; width: 100%; position: absolute; z-index: 1;"
    />
    <div
      class="container text-center"
      style="z-index: 100;"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        TeaTime
      </h1>
      <p
        class="lead"
      >
        All the teas in one, convenient place.
      </p>
      <h3>
        Welcome back, Test!
      </h3>
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="Homepage"
  >
    <img
      alt="teacup"
      src="https://images.unsplash.com/photo-1503933190776-da31c7700f80?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=3307&q=80"
      style="height: 100%; width: 100%; position: absolute; z-index: 1;"
    />
    <div
      class="container text-center"
      style="z-index: 100;"
    >
      <h1
        class="mb-4 font-weight-bold"
      >
        TeaTime
      </h1>
      <p
        class="lead"
      >
        All the teas in one, convenient place.
      </p>
      <p>
        <a
          class="btn btn-primary font-weight-bold mr-3"
          href="/login"
        >
          Log in
        </a>
        <a
          class="btn btn-primary font-weight-bold"
          href="/signup"
        >
          Sign up
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
